custom/covid/management/commands/update_cases.py

```diff
@@ -13,58 +18,66 @@ class CaseUpdateCommand(BaseCommand):
-        print(f"Found {len(case_ids)} {case_type} cases in {domain}")
-        return case_ids
-
-    def log_data(self, domain, command, case_type, total_cases, num_updated, errors, loc_id=None):
-        if self.output_file is not None:
-            with open(self.output_file, "a") as output_file:
-                num_case_updated_str = "{} {}: Updated {} out of the {} {} cases".format(domain, command,
-                                                                                         num_updated, total_cases,
-                                                                                         case_type)
-                if loc_id is not None:
-                    num_case_updated_str += f" in this location:{loc_id}"
-                output_file.write(num_case_updated_str + '\n')
-                for error in errors:
-                    output_file.write(domain + ": " + error + '\n')
+    def find_case_ids(self, domain):
+        return CommCareCase.objects.get_case_ids_in_domain(domain, self.case_type)
 
     def add_arguments(self, parser):
         parser.add_argument('domain')
         parser.add_argument('case_type')
         parser.add_argument('--username', type=str, default=None)
         parser.add_argument('--and-linked', action='store_true', default=False)
-        parser.add_argument('--output-file', type=str, default=None)
+        parser.add_argument('--throttle-secs', type=float, default=0)
 
     def handle(self, domain, case_type, **options):
+        # logger.debug will record something to a file but not print it
+        self.logger = logging.getLogger(self.logger_name())
+        if not settings.UNIT_TESTING:
+            self.logger.addHandler(logging.FileHandler(self.logger_name().split(".")[-1] + ".txt"))
+        self.logger.setLevel(logging.DEBUG)
+
+        self.logger.debug(f"{datetime.datetime.utcnow()} Starting run: {options}")
         domains = {domain}
-        if options["and_linked"]:
+        if options.pop("and_linked", False):
             domains = domains | {link.linked_domain for link in get_linked_domains(domain)}
 
-        if options["username"]:
+        username = options.pop("username", None)
+        if username is not None:
             user_id = username_to_user_id(options["username"])
             if not user_id:
                 raise Exception("The username you entered is invalid")
         else:
             user_id = SYSTEM_USER_ID
 
-        self.output_file = options["output_file"]
+        self.case_type = case_type
+        self.throttle_secs = options.pop("throttle_secs", None)
 
-        options.pop("and_linked")
-        options.pop("username")
-        options.pop("output_file", None)
         self.extra_options = options
 
-        for domain in sorted(domains):
-            print(f"Processing {domain}")
-            self.update_cases(domain, case_type, user_id)
+        username = user_id_to_username(user_id)
```

Comment:
You can delete this line and all the user_id stuff above.  The dev running the command passes in a username, and we convert it to a user ID and then back to a username here. `SystemFormMeta.for_script` does the validation as well.

Reply:
42d81f50469